### PR TITLE
Implement prefetchr/prefetchw for msvc

### PR DESCRIPTION
**Pull request by pps83**

### CODE REVIEW

File: common_defs.h [hunk 1]
```diff
@@ -203,13 +203,29 @@ typedef size_t machine_word_t;
...
 #if defined(__GNUC__) || __has_builtin(__builtin_prefetch)
 #  define prefetchr(addr)	__builtin_prefetch((addr), 0)
+#elif defined(_MSC_VER)
+#  if defined(ARCH_X86_32) || defined(ARCH_X86_64)
+#    define prefetchr(addr)	_mm_prefetch((addr), _MM_HINT_T0)
+#  elif defined(ARCH_ARM64)
+#    define prefetchr(addr)	__prefetch2((addr), 0/*PLDL1KEEP*/)
+#  elif defined(ARCH_ARM32)
+#    define prefetchr(addr)	__prefetch((addr))
+#  endif
```
Comment (ebiggers) [inline common_defs.h:213]: This is missing an `#else` case.

I'd do:

```
#undef prefetchr
(define for gcc/clang or MSVC if possible)
#ifndef prefetchr                                                                
#  define prefetchr(addr)                                                        
#endif
```

Likewise for `prefetchw`.

File: common_defs.h [hunk 2]
```diff
@@ -203,13 +203,29 @@ typedef size_t machine_word_t;
 /* prefetchr(addr) - prefetch into L1 cache for read */
 #if defined(__GNUC__) || __has_builtin(__builtin_prefetch)
 #  define prefetchr(addr)	__builtin_prefetch((addr), 0)
+#elif defined(_MSC_VER)
+#  if defined(ARCH_X86_32) || defined(ARCH_X86_64)
+#    define prefetchr(addr)	_mm_prefetch((addr), _MM_HINT_T0)
+#  elif defined(ARCH_ARM64)
+#    define prefetchr(addr)	__prefetch2((addr), 0/*PLDL1KEEP*/)
+#  elif defined(ARCH_ARM32)
+#    define prefetchr(addr)	__prefetch((addr))
```
Comment (ebiggers) [inline common_defs.h:212]: One-argument macros like `__prefetch` don't need the extra set of parantheses.

**Comment by pps83** [comment]
updated

**Comment by ebiggers** [comment]
Merged, thanks!

**Pull request merged**